integ/checkpoint_failure_test.go

```diff
@@ -85,3 +86,62 @@ func (s *IntegSuite) TestCheckpointTimeout(c *C) {
 	})
 	log.Info("Completed test TestCheckpointTimeout")
 }
+
+func (s *IntegSuite) TestSavepointCheckpointFailureFallback(c *C) {
+	log.Info("Starting test TestSavepointCheckpointFailureFallback")
+	// create a Flink app
+	testName := "recoveryfallback"
+	config, err := s.Util.ReadFlinkApplication("test_app.yaml")
+	c.Assert(err, IsNil, Commentf("Failed to read test app yaml"))
+	config.Name = testName + "job"
+	config.Spec.DeleteMode = v1beta1.DeleteModeForceCancel
+	config.Spec.FallbackWithoutState = true
+	config.ObjectMeta.Labels["integTest"] = testName
+
+	c.Assert(s.Util.CreateFlinkApplication(config), IsNil,
+		Commentf("Failed to create flink application"))
+
+	c.Assert(s.Util.WaitForPhase(config.Name, v1beta1.FlinkApplicationRunning, v1beta1.FlinkApplicationDeployFailed), IsNil)
+
+	// Cause it to fail
+	err = s.Util.ExecuteCommand("minikube", "ssh", "touch /tmp/checkpoints/fail && chmod 0644 /tmp/checkpoints/fail")
+	c.Assert(err, IsNil)
+
+	// wait a bit for it to start failing
```

Comment:
Is this comment incorrect because on line 115 you assert error is nil?

Reply:
Oh! You can not verifying that the app is failing is it?

Reply:
Lines have moved around since but if you are referring to `s.Util.GetFlinkApplication` this gets the k8s FlinkApplication CR and the err corresponds to if there were errors retrieving the CR object as opposed to anything related to the status of the actual job. 

Ln 135 `s.Util.FlinkAPIGet(newApp, endpoint)` gets the actual flink job and corresponding status to show that the job itself is healthy